packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -596,3 +619,42 @@ class FileDownloader(
     }
   }
 }
+
+private interface FileDownloadProgressListener {
+  fun update(bytesRead: Long, contentLength: Long) {
+    // Only emit progress if content length is known
+    if (contentLength > 0) {
+      onProgressUpdate(bytesRead.toDouble() / contentLength.toDouble())
+    }
+  }
+
+  fun onProgressUpdate(progress: Double) {}
+}
```

Comment:
```suggestion
private fun interface FileDownloadProgressListener {
  fun update(bytesRead: Long, contentLength: Long) {
    // Only emit progress if content length is known
    if (contentLength > 0) {
      onProgressUpdate(bytesRead.toDouble() / contentLength.toDouble())
    }
  }

  fun onProgressUpdate(progress: Double)
}
```

Reply:
We can use a SAM interface here to clean up the call site

Reply:
TIL. Thanks Alan! Updated the PR